src/diffpy/utils/tools.py

```diff
@@ -214,6 +217,49 @@ def get_package_info(package_names, metadata=None):
     return metadata
 
 
+def compute_density_from_cif(sample_composition, cif_data):
+    """Compute the theoretical density from given CIF metadata.
+
+    Parameters
+    ----------
+    sample_composition : str
+        The chemical formula of the material, e.g. "NaCl".
+    cif_data : dict
```

Comment:
We should think about this.  I wonder if it makes sense that we carry material data around always in structure object.  We could use `diffpy.structure`, but if we are moving in that direction we could use ASE structure containers.

Then we would like to pass around always whatever our "standard" structure object is and we get rid of mention of cif everywhere.   We then would have loaders that loaded cif from file into our structure container (SC), and that loaded COD-json into SC, and pymatgent whatever int SC and so on.  Since this function is, given a structure, calculate a density it should be taking our base structure object and returning density (it doesn't need to take chemical formula obviously since this is in the SC